lib/spack/spack/mirrors/utils.py

```diff
@@ -132,7 +131,8 @@ def mirror_cache_and_stats(path, skip_unstable_versions=False):
             ``fetch_strategy.stable_target``)
     """
     # Get the absolute path of the root before we start jumping around.
-    if not os.path.isdir(path):
+    path = Path(path)
+    if not path.is_dir():
```

Comment:
This is wrong if the dir exists but you don't have read permission.

Previously we'd hit the `raise MirrorError` branch, now it raises `OSError`.

That's a regression.